{{cookiecutter.project_name}}/{{cookiecutter.package_name}}/__init__.py

```diff
@@ -3,93 +3,117 @@
     schemas = {}
-
     for filename in os.listdir(get_abs_path('schemas')):
         path = get_abs_path('schemas') + '/' + filename
         file_raw = filename.replace('.json', '')
         with open(path) as file:
-            schemas[file_raw] = json.load(file)
-
+            schemas[file_raw] = Schema.from_dict(json.load(file))
     return schemas
 
+
 def discover():
     raw_schemas = load_schemas()
     streams = []
-
     for schema_name, schema in raw_schemas.items():
-
         # TODO: populate any metadata and stream's key properties here..
         stream_metadata = []
-        stream_key_properties = []
-
-        # create and add catalog entry
-        catalog_entry = {
-            'stream': schema_name,
-            'tap_stream_id': schema_name,
-            'schema': schema,
-            'metadata' : [],
-            'key_properties': []
-        }
-        streams.append(catalog_entry)
-
-    return {'streams': streams}
-
-def get_selected_streams(catalog):
-    '''
-    Gets selected streams.  Checks schema's 'selected' first (legacy)
-    and then checks metadata (current), looking for an empty breadcrumb
-    and mdata with a 'selected' entry
-    '''
-    selected_streams = []
-    for stream in catalog.streams:
-        stream_metadata = metadata.to_map(stream.metadata)
-        # stream metadata will have an empty breadcrumb
-        if metadata.get(stream_metadata, (), "selected"):
-            selected_streams.append(stream.tap_stream_id)
+        key_properties = []
+        streams.append(
+            CatalogEntry(
+                tap_stream_id=schema_name,
+                stream=schema_name,
+                schema=schema,
+                key_properties=key_properties,
+                metadata=stream_metadata,
+                replication_key=None,
+                is_view=None,
+                database=None,
+                table=None,
+                row_count=None,
+                stream_alias=None,
+                replication_method=None,
+            )
+        )
+    return Catalog(streams)
+
+
+def read_all_data():
+    """ Read all rows from upstream source and return data in proper json format """
+    # TODO: Replace with real logic to retrieve data
+    for dummy_value in range(10000):
+        yield {"id": dummy_value, "name": str(dummy_value)}
+
+
+def get_row_batches(batch_size=1):
+    """ Return lists of data rows according to batch_size """
+    queued = []
+    queued_count = 0
+    for item in read_all_data():
+        queued.append(item)
+        queued_count += 1
+        if queued_count >= batch_size:
+            yield queued
+            queued = []
+            queued_count = 0
+    if queued:
+        yield queued
 
-    return selected_streams
 
 def sync(config, state, catalog):
-
-    selected_stream_ids = get_selected_streams(catalog)
-
+    """ Sync data from tap source """
+    selected_stream_ids = catalog.get_selected_streams(state)
```

Comment:
Looks like this is unused? I think the `get_selected_streams` call should return the `CatalogEntry` objects, so you could probably rename this and replace `catalog.streams` in the for loop with it.

Reply:
Yes, that sounds right. Good catch! I'll make the change...

Reply:
updated. thanks!